code_01_dataprep_bno.R

```diff
@@ -0,0 +1,204 @@
+                "Pomatorhinus horsfieldii",
+                "Ficedula nigrorufa",
+                "Pycnonotus jocosus",
+                "Iole indica",
+                "Hemipus picatus",
+                "Saxicola caprata",
+                "Eumyias albicaudatus",
+                "Rhopocichla atriceps")) %>% 
+  auk_country(country = "IN") %>%
+  auk_state(c("IN-KL","IN-TN", "IN-KA")) %>% # Restricting geography to TamilNadu, Kerala & Karnataka
+  auk_date(c("2000-01-01", "2018-09-17")) %>% 
+  auk_complete()
+
+#'check filters
+ebd_filters
+
+#'run filters and write
+#'NB: this is already done, skip this step
+#'
+
+f_out_ebd <- "ebd_Filtered_May2018.txt"
+f_out_sampling <- "ebd_sampling_Filtered_May2018.txt"
+
+# Below code need not be run if it has been filtered once already and the above path leads to
+# the right dataset 
+# ebd_filtered <- auk_filter(ebd_filters, file = f_out_ebd, 
+#                           file_sampling = f_out_sampling)
+
+#### read the ebird data in ####
+ebd <- read_ebd(f_out_ebd)
+glimpse(ebd)
+
+#### fill zeroes ####
+zf <- auk_zerofill(f_out_ebd, f_out_sampling)
+new_zf <- collapse_zerofill(zf) # Creates a new zero-filled dataframe with a 0 marked for each checklist when the bird was not observed
+#   glimpse(zf)
+
+#### remove old data ####
+rm(data, ebd_filters, ebd_subset, zf, zf_subset)
+
+#### subset data, choose black and orange flycatcher ####
+columnsOfInterest = c("checklist_id","scientific_name","observation_count","locality","locality_id","locality_type","latitude","longitude","observation_date","time_observations_started","observer_id","sampling_event_identifier","protocol_type","duration_minutes","effort_distance_km","effort_area_ha","number_observers","species_observed","reviewed")
+
+speciesOfInterest = c("Ficedula nigrorufa")
+
+data = list(ebd, new_zf) %>% 
+  map(function(x){
+    x %>% select(one_of(columnsOfInterest)) %>% 
+      filter(scientific_name %in% speciesOfInterest)
+    # %>% dlply("scientific_name")
+    #'this above for when there are more species
+  })
+
+#### filter spatially ####
+#'load shapefiles of hill ranges
+library(sf)
+hills = st_read("hillsShapefile/Nil_Ana_Pal.shp")
+#'convert data to sf and filter spatially
+data = data %>% 
+  map(function(x){
+    x %>% st_as_sf(coords = c("longitude","latitude")) %>% 
+      st_set_crs(4326)
+  }) %>% 
+  map(function(x){
+    st_intersection(hills, x)
+  })
+
+#### presence and absence ####
+#'save a temp data file
+#save(data, file = "data.temp.rdata")
+
+load("data.temp.rdata")
+  
+#'check presence and absence in absences df, remove essentially the presences df
+data[[2]] = data[[2]] %>% filter(species_observed == F)
+
+#'in the first set, replace X, for presences, with 1
+data[[1]] = data[[1]] %>% mutate(observation_count = ifelse(observation_count == "X", "1", observation_count))
+
+#'remove records where duration is 0 and effort was > 0
+data = map(data, function(x) filter(x, duration_minutes > 0))
```

Comment:
Just to confirm, this keeps only those records whose duration is > 0 right?

Reply:
yes